idunn/directions/client.py

```diff
@@ -0,0 +1,102 @@
+import requests
+import logging
+from idunn import settings
+from datetime import datetime, timedelta
+from .models import DirectionsResponse
+from apistar.http import JSONResponse
+from apistar.exceptions import BadRequest, HTTPException
+
+logger = logging.getLogger(__name__)
+
+
+class DirectionsClient:
+    def __init__(self):
+        self.qw_session = requests.Session()
+        self.qw_session.verify = False
+        self.qw_session.headers['User-Agent'] = 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:68.0) Gecko/20100101 Firefox/68.0'
+
+        self.combigo_session = requests.Session()
+        self.combigo_session.headers['x-api-key'] = settings['COMBIGO_API_KEY']
+
+    @property
+    def QWANT_BASE_URL(self):
+        return settings['QWANT_DIRECTIONS_API_BASE_URL']
+
+    @property
+    def COMBIGO_BASE_URL(self):
+        return settings['COMBIGO_API_BASE_URL']
+
+    def directions_qwant(self, start, end, mode, lang, extra=None):
+        if not self.QWANT_BASE_URL:
+            raise HTTPException(f"Directions API is currently unavailable for mode {mode}", status_code=501)
+
+        if extra is None:
+            extra = {}
+        start_lon, start_lat = start
+        end_lon, end_lat = end
+        response = self.qw_session.get(
+            f'{self.QWANT_BASE_URL}/{start_lon},{start_lat};{end_lon},{end_lat}',
+            params={
+                'type': mode,
+                'language': lang,
+                'steps': 'true',
+                'alternatives': 'true',
+                'overview': 'full',
+                'geometries': 'geojson',
+                **extra
+            }
+        )
+
+        if 400 <= response.status_code < 500:
```

Comment:
TIL

Reply:
Isn't there a way to just check on the status code directly it's an error one?

Reply:
no, it does not seem that "requests" library provides any other method for that